ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -0,0 +1,160 @@
+                || FUNC_NAME == FUNC_NAMES.ARRAY_JOIN) {
+            checkArgCategory(arguments, ARRAY2_IDX, ObjectInspector.Category.LIST, FUNC_NAME,
+                    org.apache.hadoop.hive.serde.serdeConstants.LIST_TYPE_NAME);
+        }
+
+        if (FUNC_NAME == FUNC_NAMES.ARRAY_SLICE) {
+            PrimitiveObjectInspector startIndexObjectInspector = (PrimitiveObjectInspector) arguments[START_IDX];
+            PrimitiveObjectInspector lengthObjectInspector = (PrimitiveObjectInspector) arguments[LENGTH_IDX];
+            checkArgIntPrimitiveCategory(startIndexObjectInspector, FUNC_NAME, 2);
+            checkArgIntPrimitiveCategory(lengthObjectInspector, FUNC_NAME, 3);
+        }
+
+        arrayOI = (ListObjectInspector) arguments[ARRAY_IDX];
+        argumentOIs = arguments;
+
+        //return initialize(arguments);
+        return initListOI(arguments);
+    }
+
+    @Override
+    public String getDisplayString(String[] children) {
+        assert (children.length == MIN_ARG_COUNT);
+        return FUNC_NAME.toString().toLowerCase() + "(" + children[ARRAY_IDX] + ")";
+    }
+
+    List<Object> convertArray(List objects) {
+        List<Object> ret = new ArrayList<>();
+        for (Object o : objects) {
+            ret.add(converter.convert(o));
+        }
+        return ret;
+    }
+
+    void checkArgCategory(ObjectInspector[] arguments, int idx, Enum category,
+                          FUNC_NAMES function_name, String typeName) throws UDFArgumentTypeException {
+
+        if (!arguments[idx].getCategory().equals(category)) {
+            throw new UDFArgumentTypeException(idx,
+                    "\"" + typeName + "\" "
+                            + "expected at function " + function_name + ", but "
+                            + "\"" + arguments[idx].getTypeName() + "\" "
+                            + "is found");
+        }
+    }
+
+    void checkArgIntPrimitiveCategory(PrimitiveObjectInspector objectInspector,
+                                      FUNC_NAMES function_name, int idx) throws UDFArgumentTypeException {
+
+        switch (objectInspector.getPrimitiveCategory()) {
+            case SHORT:
+            case INT:
+            case LONG:
+                break;
+            default:
+                throw new UDFArgumentTypeException(0, "Argument " + idx
+                        + " of function " + function_name + " must be \""
+                        + serdeConstants.SMALLINT_TYPE_NAME + "\""
+                        + " or \"" + serdeConstants.INT_TYPE_NAME + "\""
+                        + " or \"" + serdeConstants.BIGINT_TYPE_NAME + "\", but \""
+                        + objectInspector.getTypeName() + "\" was found.");
+        }
+    }
+
+    boolean isListEmpty(Object array, ListObjectInspector listObjectInspector) {
+
+        int arrayLength = listObjectInspector.getListLength(array);
+
+        // Check if array is null or empty or value is null
+        return array == null || arrayLength <= 0;
```

Comment:
optional nit: I would prefer this to be one line, as in 
return array == null || listObjectInspector.getListLength(array) <= 0;

...and then get rid of all the blank lines above and below

Reply:
This can be simplified to -
`return listObjectInspector.getListLength(array) <= 0;`

Because the function `getListLength` is already checking for null values within and returning -1. See here - 
https://github.com/apache/hive/blob/master/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ListObjectInspector.java#L42-L45